test/integration/test_save_job_id_on_datasets.py

```diff
@@ -0,0 +1,42 @@
+"""
```

Comment:
Can you add a basic test somewhere around there as well ? https://github.com/mvdbeek/galaxy/blob/65479c5cd6981ee6d86077a38369af2c0421d4a2/test/unit/data/test_galaxy_mapping.py#L382

Reply:
Do you mean a test of the dataset/job relationship? Like
- create dataset, create job, set dataset.job = job, then persist, then load dataset by id, then assert that dataset.job_id == job.id? I'll commit this in a min - let me know if you wanted me to test something else.